databases/ClickHouse/Portfile

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github  1.0
+PortGroup           cmake   1.1
+
+github.setup        ClickHouse ClickHouse 22.4.2.1 v -stable
+
+description         ClickHouse® is a free analytics DBMS for big data
+
+long_description \
+    ClickHouse® is an open-source column-oriented database management \
+    system that allows generating analytical data reports in real-time.
+
+notes {
+    Install the ClickHouse-server port to run ClickHouse as a system service.
+}
+
+categories          databases
+maintainers         @slarew openmaintainer
+license             Apache-2
+platforms           darwin
+
+# clickhouse overrides the deployment target so match it
+macosx_deployment_target 10.15
```

Comment:
So then this port cannot possibly work on macOS 10.14 and earlier, so you should have a block that sets `known_fail yes` in that case and prints a message to the user. See other ports for examples.